src/compile/typecheck.rs

```diff
@@ -1613,13 +1624,19 @@ fn typecheck_expr(
                 )?;
                 if tc_sub1.get_type() != Type::Bool {
                     return Err(new_type_error(
-                        "operands to logical or must be boolean".to_string(),
+                        format!(
```

Comment:
It seems better to print both operand types here, so the user doesn't have to guess which one has the wrong type.